controllers/flinkcluster_reconciler.go

```diff
@@ -468,27 +468,24 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 		var jobID = reconciler.getFlinkJobID()
 		var restartPolicy = observed.cluster.Spec.Job.RestartPolicy
 		var recordedJobStatus = observed.cluster.Status.Components.Job
+		var jobSpec = reconciler.observed.cluster.Spec.Job
 
 		// Update or recover Flink job by restart.
-		var restartJob bool
 		if shouldUpdateJob(observed) {
 			log.Info("Job is about to be restarted to update")
-			restartJob = true
+			err := reconciler.restartJob(*jobSpec.ShouldTakeSavepointOnUpgrade)
```

Comment:
I prefer keeping the existing structure and introducing a variable `takeSavepointOnUpdate` for the 2 cases.

Reply:
You sure? each case needs to pass a different argument to restartJob. I think its much cleaner like that